jerry-ext/module/module.c

```diff
@@ -180,29 +193,60 @@ jerryx_module_native_resolver (const jerry_char_t *name, /**< name of the module
  *   - an error indicating that something went wrong during the attempt to load the module.
  */
 jerry_value_t
-jerryx_module_resolve (const jerry_char_t *name, /**< name of the module to load */
-                       const jerryx_module_resolver_t *resolvers_p, /**< list of resolvers */
-                       size_t resolver_count) /**< number of resolvers in @p resolvers */
+jerryx_module_resolve (const jerry_value_t name, /**< name of the module to load */
+                       const jerryx_module_resolver_t **resolvers_p, /**< list of resolvers */
+                       const jerry_size_t count) /**< length of list */
 {
   size_t index;
-  jerry_value_t ret;
-  jerry_value_t instances = *(jerry_value_t *) jerry_get_context_data (&jerryx_module_manager);
-  jerry_value_t module_name = jerry_create_string_from_utf8 (name);
+  jerry_value_t ret = 0;
+  const jerryx_module_resolver_t *resolver_p;
+  jerry_value_t instances;
+  jerry_value_t canonical_names_static[5];
```

Comment:
Lets use a define for this.

Reply:
What do you mean "use a define"?

Reply:
`#define CANONICAL_NAMES_LOCAL_SIZE 4` or something. I think 4 is enough (probably 2 is the most frequent in practice).